lib/model.js

```diff
@@ -63,6 +63,11 @@ ModelBaseClass.prototype._initProperties = function (data, options) {
   var self = this;
   var ctor = this.constructor;
 
+  if (typeof data !== 'undefined' && 
+     typeof (data.constructor) !== 'function') {
+    throw new Error('Property names could not be constructor ');
```

Comment:
Please remove the trailing space from the message.

I am proposing to reword the message a bit:

```js
throw new Error('Property name "constructor" is not allowed in ' + ctor.modelName +' data');
```